### ISSUE TITLE
Allow to clone unsigned extrinsic

### ISSUE THREAD

**Issue by ianhe8x** [polkadot-js/api#1274]
Currently this test will fail.
```
it('clone an unsigned extrinsic', (): void => {
      const extrinsic = extrinsics.timestamp.set(new Date());
      const clone = new Extrinsic(extrinsic.toU8a());
      expect(clone.isSigned).toEqual(false);
      expect(clone.encodedLength).toEqual(extrinsic.encodedLength);
      expect(clone.toHex()).toEqual(extrinsic.toHex());
    });
```
From error msg it's because in packages/types/src/primitive/Extrinsic/Extrinsic.ts there's a check of compact length, when the extrinsic is unsigned, the length won't match (because the absence of signature?).

```
ExtError: Extrinsic: required length less than remainder, expected at least 4, found 3

    at assert (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/node_modules/@polkadot/util/assert.js:47:9)
    at Function.decodeExtrinsic (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/packages/types/src/primitive/Extrinsic/Extrinsic.ts:85:7)
    at new decodeExtrinsic (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/packages/types/src/primitive/Extrinsic/Extrinsic.ts:41:21)
    at Object.it (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/packages/types/src/primitive/Extrinsic/Extrinsic.spec.ts:73:21)
    at Object.asyncJestTest (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/Users/moge/Workspace/centrality/cennznet/polkadot-js-api/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by jacogr** [comment]
`const extrinsic = extrinsics.timestamp.set(new Date());` does not return an extrinsic, rather it is a `Call`. (The first element, after the version of an extrinsic). so the encoded version there does not have length nor the actual version identifier.

If you wish to construct via `Call`, you would need to -

- pass in a `Call` instance, i.e. `new Extrinsic(createType('Call', x.toU8a()), { version: 3 })`
- As per above, doing this also requires passing in the actual extrinsic version, either 1 (original), 2 (with tip) or 3 (to sign with genesisHash as well)

**Comment by ianhe8x** [comment]
the `const extrinsic = extrinsics.timestamp.set(new Date())` thing is just for explaining my issue in a simplified way. My problem is for extrinsic created via api.tx.xxx.xxx(), and i figure out it's a problem in v0.82.x only. 
in v1/ExtrinsicSignature.ts, before 0.90, for unsigned extrinsic, encodedLength is not 0, which is fixed in #1216.
